packages/expo-cli/src/appleApi/authenticate.ts

```diff
@@ -111,22 +117,53 @@ async function _promptForAppleId({
 
     // https://docs.expo.io/distribution/security/#apple-developer-account-credentials
     const here = terminalLink('here', 'https://bit.ly/2VtGWhU');
-    log(wrap(chalk.bold(`The password is only used to authenticate with Apple and never stored`)));
+    log(
+      wrap(
+        chalk.bold(
+          `The password is only used to authenticate with Apple and never stored on Expo servers`
+        )
+      )
+    );
     log(wrap(chalk.grey(`Learn more ${here}`)));
   }
 
+  // Get the email address that was last used and set it as
+  // the default value for quicker authentication.
+  const lastAppleId = await getLastUsedAppleIdAsync();
+
   const { appleId: promptAppleId } = await prompt(
     {
       type: 'input',
       name: 'appleId',
       message: `Apple ID:`,
       validate: nonEmptyInput,
+      default: lastAppleId as string,
       ...(previousAppleId && { default: previousAppleId }),
     },
     {
       nonInteractiveHelp: 'Pass your Apple ID using the --apple-id flag.',
     }
   );
+
+  // If a new email was used then store it as a suggestion for next time.
+  // This functionality is disabled using the keychain mechanism.
+  if (!Keychain.EXPO_NO_KEYCHAIN && lastAppleId !== promptAppleId) {
```

Comment:
if someone doesn't have `appleId` set in UserSettings, and they press `ENTER`, appleId would be `null`. Let's only store `lastAppleId` if its truthy
```suggestion
  if (!Keychain.EXPO_NO_KEYCHAIN && lastAppleId && lastAppleId !== promptAppleId) {
```